internal/backend/temporalclient/largepayloads_codec.go

```diff
@@ -0,0 +1,178 @@
+package temporalclient
+
+import (
+	"errors"
+	"maps"
+	"path/filepath"
+	"strconv"
+
+	"github.com/philippgille/gokv"
+	"github.com/philippgille/gokv/encoding"
+	"github.com/philippgille/gokv/file"
+	"github.com/philippgille/gokv/gomap"
+	"github.com/philippgille/gokv/s3"
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/internal/xdg"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+const (
+	largeObjectKeyMetadataKey  = "ak_lo_key"
+	largeObjectSizeMetadataKey = "ak_lo_size"
+)
+
+type LargePayloadConfig struct {
+	StoreType string `koanf:"store_type"` // "none" to disable.
+
+	// ThresholdSize is the maximum size of a payload that will be stored in Temporal.
+	// If a payload is larger than this size, it will be stored in a separate store.
+	ThresholdSize int `koanf:"threshold_size"`
+
+	FileStoreRootPath string `koanf:"file_store_root_path"` // if store_type == "file".
+	S3StoreBucketName string `koanf:"s3_store_bucket_name"` // if store_type == "s3".
```

Comment:
Nit: S3 = "Simple Storage Service". So "S3 store" is like saying "ATM machine".

Actually, why not use just one string for the local-files root path and cloud-storage bucket URL?

Reply:
changed names, but i think explicit fields are better.